src/elements/content-explorer/__tests__/ContentExplorer-test.js

```diff
@@ -42,16 +43,113 @@ describe('elements/content-explorer/ContentExplorer', () => {
+        });
+
         test('should initially be list view', () => {
-            const wrapper = getWrapper();
             expect(wrapper.state('viewMode')).toBe(VIEW_MODE_LIST);
         });
 
         test('should change to grid view', () => {
-            const wrapper = getWrapper();
             const instance = wrapper.instance();
             instance.changeViewMode(VIEW_MODE_GRID);
             expect(wrapper.state('viewMode')).toBe(VIEW_MODE_GRID);
         });
     });
+
+    describe('fetchFolder()', () => {
+        const getFolder = jest.fn();
+        const getFolderAPI = jest.fn().mockReturnValue({
+            getFolder,
+        });
+
+        let wrapper;
+        let instance;
+
+        test('should fetch folder without representations field if grid view is not enabled', () => {
+            wrapper = getWrapper();
+            instance = wrapper.instance();
+            instance.api = { getFolderAPI };
+            instance.setState = jest.fn();
+            instance.fetchFolder();
+            expect(instance.setState).toHaveBeenCalled();
+            expect(getFolder).toHaveBeenCalledWith(
+                '0',
+                50,
+                0,
+                'name',
+                'ASC',
+                expect.any(Function),
+                expect.any(Function),
+                { forceFetch: true, fields: FOLDER_FIELDS_TO_FETCH },
+            );
+        });
+
+        test('should fetch folder with representations field if grid view is enabled', () => {
+            wrapper = getWrapper({ features: { contentExplorer: { gridView: { enabled: true } } } });
+            instance = wrapper.instance();
+            instance.api = { getFolderAPI };
+            instance.setState = jest.fn();
+            instance.fetchFolder();
+            expect(instance.setState).toHaveBeenCalled();
+            expect(getFolder).toHaveBeenCalledWith(
+                '0',
+                50,
+                0,
+                'name',
+                'ASC',
+                expect.any(Function),
+                expect.any(Function),
+                { forceFetch: true, fields: [...FOLDER_FIELDS_TO_FETCH, FIELD_REPRESENTATIONS] },
+            );
+        });
+    });
+
+    describe('fetchFolderSuccessCallback()', () => {
+        const item = { id: 1 };
+        const collection = { boxItem: {}, id: '0', items: [item], name: 'name' };
+        const thumbnailUrl = 'thumbnailUrl';
+        const getThumbnailUrl = jest.fn().mockReturnValue(thumbnailUrl);
+        const getFileAPI = jest.fn().mockReturnValue({
+            getThumbnailUrl,
+        });
+
+        let wrapper;
+        let instance;
+
+        test('thumbnail url should not be assigned to item if grid view is not enabled', () => {
+            wrapper = getWrapper();
+            instance = wrapper.instance();
+            instance.api = { getFileAPI };
+            instance.setState = jest.fn();
+            instance.closeModals = jest.fn();
```

Comment:
Is it useful it to check functions like this even though they are not specific to what is being tested?

Reply:
It can be, as an indication of the code paths which are being executed under test. In this case, maybe not so much since it's always called no matter what. But stubbing I would still recommend because we wouldn't want any side effects of calling that method to affect our test